src/main/java/com/coveros/selenified/element/check/State.java

```diff
@@ -0,0 +1,314 @@
+        // perform the check
+        boolean isPresent = getElement().is().present();
+        // record the result
+        if (isPresent) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + PRESENT, timeTook, Success.PASS);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + NOTPRESENT, timeTook, Success.FAIL);
+        }
+        return isPresent;
+    }
+
+    /**
+     * Verifies that the element is not present. If the element is present, it
+     * waits up to the default time (5 seconds) for the element to be removed,
+     * before marking this verification as a failure. This information will be
+     * logged and recorded, with a screenshot for traceability and added
+     * debugging support.
+     */
+    void notPresent();
+
+    default boolean checkNotPresent(double waitFor, double timeTook) {
+        // record the action
+        getOutputFile().recordAction(getElement().prettyOutput() + NOTPRESENT, waitFor);
+        // perform the check
+        boolean isPresent = getElement().is().present();
+        // record the result
+        if (isPresent) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + PRESENT, timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + NOTPRESENT, timeTook, Success.PASS);
+        }
+        return !isPresent;
+    }
+
+    /**
+     * Verifies that the element is displayed. If the element isn't present, it
+     * waits up to the default time (5 seconds) for the element, before marking
+     * this verification as a failure. This information will be logged and
+     * recorded, with a screenshot for traceability and added debugging support.
+     */
+    void displayed();
+
+    default boolean checkDisplayed(double waitFor, double timeTook) {
+        // record the action
+        getOutputFile().recordAction(getElement().prettyOutput() + DISPLAYED, waitFor);
+        // verify the element is present
+        if (!isPresent(waitFor)) {
+            return false;
+        }
+        // perform the check
+        boolean isDisplayed = getElement().is().displayed();
+        // record the result
+        if (!isDisplayed) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + NOTDISPLAYED, timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + DISPLAYED, timeTook, Success.PASS);
+        }
+        return isDisplayed;
+    }
+
+    /**
+     * Verifies that the element is not displayed. If the element isn't present,
+     * it waits up to the default time (5 seconds) for the element, before
+     * marking this verification as a failure. This information will be logged
+     * and recorded, with a screenshot for traceability and added debugging
+     * support.
+     */
+    void notDisplayed();
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Rename method "notDisplayed" to prevent any misunderstanding/clash with field "NOTDISPLAYED" defined on line 43. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1845)